[iOS] ToolbarItem color with custom BarTextColor - fix

kubaflo opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Description of Change

This PR ensures that toolbar items use the correct tint color when `BarTextColor` is set by updating their image rendering mode to `Automatic`. This addresses issues where UIBarButtonItem images would not reflect the intended color due to AlwaysOriginal rendering mode.

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes https://github.com/dotnet/maui/issues/30818
https://github.com/dotnet/maui/issues/30818#issuecomment-3169455559

kubaflo committed:
Fix right bar button item color with custom BarTextColor

Ensures right bar button items use the correct tint color when BarTextColor is set by updating their image rendering mode to Automatic. This addresses issues where UIBarButtonItem images would not reflect the intended color due to AlwaysOriginal rendering mode.

PureWeen commented:
/asp run

PureWeen commented on src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs:
```diff
@@ -1698,7 +1698,19 @@ void UpdateToolbarItems()
+					{
+						foreach (var item in NavigationItem.RightBarButtonItems)
+						{
+							item.Image = item.Image?.ImageWithRenderingMode(UIImageRenderingMode.Automatic);
```
Can we add the color to the "ToUIBarButtonItem" call?

Or maybe we can add a bool that will indicate whether to toggle between the different rendering modes based on if BarTextColor is set?

That way the behavior should be pretty much the same as before

kubaflo replied:
Sure, I've added a commit

kubaflo committed:
Support toolbar item color for iOS navigation bar

Adds support for passing a toolbar item color to primary toolbar items in iOS navigation bars. Updates the NavigationRenderer to use the BarTextColor for toolbar items, and modifies ToolbarItemExtensions to allow specifying a color when creating UIBarButtonItems.

PureWeen commented:
/azp run

jfversluis reviewed (changes requested):
Windows build fails

> C:\a\_work\1\s\src\Compatibility\Core\src\iOS\Renderers\NavigationRenderer.cs(1418,71): error CS0121: The call is ambiguous between the following methods or properties: 'ToolbarItemExtensions.ToUIBarButtonItem(ToolbarItem, bool, bool, Color)' and 'ToolbarItemExtensions.ToUIBarButtonItem(ToolbarItem, bool, bool)' [C:\a\_work\1\s\src\Compatibility\Core\src\Compatibility.csproj::TargetFramework=net9.0-maccatalyst18.0]
C:\a\_work\1\s\src\Compatibility\Core\src\iOS\Renderers\NavigationRenderer.cs(1418,71): error CS0121: The call is ambiguous between the following methods or properties: 'ToolbarItemExtensions.ToUIBarButtonItem(ToolbarItem, bool, bool, Color)' and 'ToolbarItemExtensions.ToUIBarButtonItem(ToolbarItem, bool, bool)' [C:\a\_work\1\s\src\Compatibility\Core\src\Compatibility.csproj::TargetFramework=net9.0-ios18.0]

PureWeen committed:
- fix compile errors

PureWeen committed:
- add tests

PureWeen committed:
- fix set color color

PureWeen reviewed (changes requested):
Looking a bit deeper at this one I'm just going to pause on this one

One other thing that broke from the original PR is that the thing now doesn't go to that "iOS" blue anymore and that's going to be a breaking behavior for a lot of projects

PureWeen commented:
@kubaflo I'm going to close this one for now until we assess the original issue a bit closer

I suspect the fix we end up with there is going to differ from this PR a good amount
